pandas/core/dtypes/cast.py

```diff
@@ -119,16 +119,21 @@
 _int64_max = np.iinfo(np.int64).max
 
 
-def maybe_convert_platform(values):
+def maybe_convert_platform(
+    values: Union[list, tuple, range, np.ndarray, ExtensionArray]
+) -> ArrayLike:
     """ try to do platform conversion, allow ndarray or list here """
     if isinstance(values, (list, tuple, range)):
-        values = construct_1d_object_array_from_listlike(values)
-    if getattr(values, "dtype", None) == np.object_:
-        if hasattr(values, "_values"):
-            values = values._values
-        values = lib.maybe_convert_objects(values)
+        arr = construct_1d_object_array_from_listlike(values)
+    else:
+        # The caller is responsible for ensuring that we have np.ndarray
```

Comment:
we should tighten this and return a np.ndarray

Reply:
im planning to continue tightening, yes.  But im also hoping to start passing `convert_datetime=True, convert_timedelta=True` to lib.maybe_convert_objects on L134.